agreementbot/agreementworker.go

```diff
@@ -852,7 +852,7 @@ func (b *BaseAgreementWorker) ValidateAndExtractSecrets(consumerPolicy *policy.P
 				}
 
 				// Call the secret manager plugin to get the secret details.
-				details, err := b.secretsMgr.GetSecretDetails(exchange.GetOrg(deviceId), secretUser, shortSecretName)
+				details, err := b.secretsMgr.GetSecretDetails(b.config.AgreementBot.ExchangeId, b.config.AgreementBot.ExchangeToken, exchange.GetOrg(deviceId), secretUser, shortSecretName)
```

Comment:
I think the more correct way is this:
GetSecretDetails(b.GetExchangeId(), b.GetExchangeToken(), ...)